src/openforms/registrations/contrib/objects_api/tests/test_template.py

```diff
@@ -305,3 +309,109 @@ def test_submission_with_objects_api_content_json_not_valid_json(self):
         ):
             with self.assertRaises(RuntimeError):
                 plugin.register_submission(submission, {})
+
+
+class JSONTemplatingRegressionTests(SubmissionsMixin, APITestCase):
+    @tag("dh-673")
+    def test_object_nulls_regression(self):
+        submission = SubmissionFactory.from_components(
+            components_list=[
+                {
+                    "type": "radio",
+                    "key": "radio",
+                    "label": "Radio",
+                    "values": [
+                        {"label": "1", "value": "1"},
+                        {"label": "2", "value": "2"},
+                    ],
+                    "defaultValue": None,
+                    "validate": {"required": True},
+                    "openForms": {"dataSrc": "manual"},
+                },
+                {
+                    "type": "textfield",
+                    "key": "tekstveld",
+                    "label": "Tekstveld",
+                    "hidden": True,
+                    "validate": {"required": True},
+                    "conditional": {"eq": "1", "show": True, "when": "radio"},
+                    "defaultValue": None,
+                    "clearOnHide": True,
+                },
+                {
+                    "type": "currency",
+                    "currency": "EUR",
+                    "key": "bedrag",
+                    "label": "Bedrag",
+                    "hidden": True,
+                    "validate": {"required": True},
+                    "conditional": {"eq": "1", "show": True, "when": "radio"},
+                    "defaultValue": None,
+                    "clearOnHide": True,
+                },
+            ],
+            with_report=True,
+            submitted_data={"radio": "2"},
+            form_definition_kwargs={"slug": "stepwithnulls"},
+        )
+        self._add_submission_to_session(submission)
+        # complete the submission
+        endpoint = reverse("api:submission-complete", kwargs={"uuid": submission.uuid})
+        with self.captureOnCommitCallbacks(execute=True):
+            response = self.client.post(endpoint, {"privacy_policy_accepted": True})
+        assert response.status_code == status.HTTP_200_OK
```

Comment:
Was this necessary to have the empty values for hidden components (as described in the comment later in this test)?

Reply:
hrm, no, in hindsight it turns out that the completion of the submission did not mutate the data in any way, which I suspected at first :thinking: